core/src/data_sources/splitter.rs

```diff
@@ -39,6 +39,12 @@ impl FromStr for SplitterID {
 pub trait Splitter {
     fn id(&self) -> SplitterID;
 
+    async fn decode_chunk_with_remainder(
```

Comment:
Should not be on the trait

Reply:
The trait is the shared interface of splitters, and this is definitely an internal implementation.

You want to add a `impl SplitterBaseV0 {...}`